homeassistant/helpers/script.py

```diff
@@ -477,8 +477,6 @@ async def async_run(self) -> ScriptRunResult | None:
 
             # Let the _StopScript bubble up if this is a sub-script
             if not self._script.top_level:
-                # We already consumed the response, do not pass it on
-                err.response = None
```

Comment:
This check was intentionally added in https://github.com/home-assistant/core/pull/95284, are we sure removing it does not have unwanted side effects?

Reply:
What you're suggesting is this change:
```diff
         except _StopScript as err:
             script_execution_set("finished", err.response)
             response = err.response
 
             # Let the _StopScript bubble up if this is a sub-script
             if not self._script.top_level:
-                # We already consumed the response, do not pass it on
-                err.response = None
                 raise
```

Looking at @ballob's PR https://github.com/home-assistant/core/pull/95284 I think it *may* be a mistake related to the somewhat confusing `script_execution_set("finished", err.response)` above which stores the result in the script trace.

Reply:
```py
        except _StopScript as err:
            script_execution_set("finished", err.response)

            # Let the _StopScript bubble up if this is a sub-script
            if not self._script.top_level:
                raise err

            response = err.response

```

Reply:
Extracted it in this PR: https://github.com/home-assistant/core/pull/131691